### PR TITLE
[WIP] Typescript port

### PR DESCRIPTION
**Pull request by ricokahler**
# This is very much a work in progress.

In fact, this code currently doesn't even compile. The reason I'm opening the pull request now is to ask questions and get feedback early before I change almost everything 😅 **though the exposed API won't change so nothing will break** if this eventually accepted.

@clauderic could you answer the questions below by the number when you have the time?

## Questions:

1. **How are touch events handled?** It seems like we're adding events to the raw container but there is no use of [`TouchEvent.changedTouches`](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent/changedTouches). For the event handler `handleStart`, I've typed the event as [`e: MouseEvent | TouchEvent`][0] but the touch events *don't* have `e.pageX` and `e.pageY` defined. [This is how I'm getting `pageX` and `pageY` from the touch events][1]. **Is this change okay?** I suspect we might have issues if both a click event and touch event are fired but this can be confirmed later.

[0]: https://github.com/ricokahler/react-sortable-hoc/blob/f9e281295b5e5f5d984746d83e3e9f82966c1db1/src/SortableContainer/index.tsx#L207
[1]: https://github.com/ricokahler/react-sortable-hoc/blob/f9e281295b5e5f5d984746d83e3e9f82966c1db1/src/SortableContainer/index.tsx#L217-L225

2. **What is the type for [`Manager.prototype.getActive`][2]?** I see that you're taking `HTMLElement`s and adding the property `sortableInfo` to them so [I've created and interface for that][4]. I also see the property `refs` on the class `Manager` holds these elements. I've typed this property as [`refs: { [collection: string]: SortableNode[] }`][5]. Now my question lies with how you use [`manager.getActive()` from `handlePress`][6]. My interface for `Manager.refs` doesn't agree with your implementation because [these `refs` don't have a `collection` or `node` property][7]. **What is the correct type for `refs`? Is it correct to use [`this.manager.getActive()` here?][8]**

[2]: https://github.com/ricokahler/react-sortable-hoc/blob/f9e281295b5e5f5d984746d83e3e9f82966c1db1/src/Manager.ts#L39
[3]: https://github.com/ricokahler/react-sortable-hoc/blob/f9e281295b5e5f5d984746d83e3e9f82966c1db1/src/SortableContainer/index.tsx#L331
[4]: https://github.com/ricokahler/react-sortable-hoc/blob/f9e281295b5e5f5d984746d83e3e9f82966c1db1/src/Manager.ts#L3-L9
[5]: https://github.com/ricokahler/react-sortable-hoc/blob/f9e281295b5e5f5d984746d83e3e9f82966c1db1/src/Manager.ts#L12
[6]: https://github.com/ricokahler/react-sortable-hoc/blob/f9e281295b5e5f5d984746d83e3e9f82966c1db1/src/SortableContainer/index.tsx#L319-L331
[7]: https://github.com/ricokahler/react-sortable-hoc/blob/f9e281295b5e5f5d984746d83e3e9f82966c1db1/src/SortableContainer/index.tsx#L331
[8]: https://github.com/ricokahler/react-sortable-hoc/blob/f9e281295b5e5f5d984746d83e3e9f82966c1db1/src/SortableContainer/index.tsx#L320

---

Let me know your thoughts, and if you live in the states, happy fourth!

### CODE REVIEW

**Comment by ricokahler** [comment]
https://github.com/clauderic/react-sortable-hoc/issues/231

File: src/Manager.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { find, sortBy } from 'lodash';
+
+export interface SortableNode extends HTMLElement {
+  sortableInfo: {
+    index: number,
+    collection: string,
+    manager: Manager,
+  }
+}
```
Comment (clauderic) [inline src/Manager.ts]: Nice 👍

File: examples/basic.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
-import React, {Component} from 'react';
-import {render} from 'react-dom';
-import {SortableContainer, SortableElement, arrayMove} from 'react-sortable-hoc';
+import React, { Component } from 'react';
+import { render } from 'react-dom';
```
Comment (clauderic) [inline examples/basic.js]: Small nitpick: Can we revert all these to have no spaces? Just a personal preference wrt linting 😅

Comment (ricokahler) [inline examples/basic.js]: @clauderic yeah we can do that. I have no preference. VS Code is my main editor and I use the shortcut to format code and that was the result. I just have to change my vs code settings somewhere.

Comment (clauderic) [inline examples/basic.js]: You can use [prettier](https://github.com/prettier/prettier) for all linting, it's what I used with the repo, with these configs:

```
 "prettier.singleQuote": true,
 "prettier.trailingComma": "es5",
 "prettier.bracketSpacing": false,
```

**Comment by clauderic** [comment]
Moving this over to #250

**Pull request closed without merging**